standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/utils/TestMetaStoreUtils.java

```diff
@@ -67,9 +69,31 @@ public static Collection<Object[]> generateZoneTimestampPairs() {
         params.add(new Object[] { zone, datetime });
       }
     });
+    // Timestamp and Date do not have the year 0000. So, the year 0000 gets converted to year 0001.
+    // This is one such test case with year as 0000: 0000-01-07 22:44:36
+    params.add(new Object[] {"Asia/Kolkata", LocalDateTime.ofEpochSecond(-62166618924L, 0, ZoneOffset.UTC)});
+    generateDaylightSavingTimestampPairs(params);
     return params;
   }
 
+  public static void generateDaylightSavingTimestampPairs(List<Object[]> params) {
+    // For U.S timezones, the timestamp 2024-03-10 02:01:00 falls between the transition from 02:00 AM to 03:00 AM for daylight savings
+    params.add(new Object[] {"America/Anchorage", LocalDateTime.ofEpochSecond(1710036060L, 0, ZoneOffset.UTC)});
```

Comment:
Instead of using the `LocalDateTime.ofEpochSecond` constructor you could use the basic constructor which allows you to specify all fields explicitly. By doing this the comment above becomes redundant since it is clear from the code what is the date.